code/ATMOSPHERICS/components/binary_devices/circulator.dm

```diff
@@ -3,76 +3,68 @@
 /obj/machinery/atmospherics/binary/circulator
 	name = "circulator/heat exchanger"
 	desc = "A gas circulator pump and heat exchanger."
-	icon = 'icons/obj/pipes.dmi'
-	icon_state = "circ-off"
-	anchored = 0
-	density = 1
+	icon = 'icons/obj/atmospherics/circulator.dmi'
+	icon_state = "circ1-off"
+
+	var/side = CIRC_LEFT
+	
+	var/global/const/CIRC_LEFT = 1
+	var/global/const/CIRC_RIGHT = 2
 
-	var/recent_moles_transferred = 0
-	var/last_heat_capacity = 0
-	var/last_temperature = 0
 	var/last_pressure_delta = 0
-	var/last_worldtime_transfer = 0
 
+	anchored = 1
+	density = 1
+	
+	can_unwrench = 1
+	
 /obj/machinery/atmospherics/binary/circulator/New()
+	initialize()

```

Comment:
I believe `initialize` is called in `New` somewhere up the tree--- @Crazylemon64 would know better.